game/scripts/vscripts/abilities/visage_gravekeepers_cloak.lua

```diff
@@ -0,0 +1,109 @@
+
+if Debug then
+  Debug.EnabledModules['abilities:*'] = true
+end
+
+LinkLuaModifier("modifier_visage_gravekeepers_cloak_oaa", "abilities/visage_gravekeepers_cloak.lua", LUA_MODIFIER_MOTION_NONE) --- PETH WEFY INPARFANT
+LinkLuaModifier("modifier_visage_gravekeepers_cloak_oaa_aura", "abilities/visage_gravekeepers_cloak.lua", LUA_MODIFIER_MOTION_NONE) --- PETH WEFY INPARFANT
+
+visage_gravekeepers_cloak_oaa = class(AbilityBaseClass)
+
+function visage_gravekeepers_cloak_oaa:GetIntrinsicModifierName()
+  return "modifier_visage_gravekeepers_cloak_oaa"
+end
+
+if IsServer() then
+  function visage_gravekeepers_cloak_oaa:OnUpgrade()
+    local caster = self:GetCaster()
+    local mod = caster:FindModifierByName("modifier_visage_gravekeepers_cloak_oaa")
+    mod:SetStackCount(self:GetSpecialValueFor("max_layers"))
```

Comment:
Oh wait, doing it only on level 1 would break the higher stacks... Seems weird to just refresh all stacks when upgrading though. Maybe it should check if the max stacks has increased compared to last level and add that many.